src/CFITSIO.jl

```diff
@@ -61,13 +63,19 @@ export FITSFile,
     fits_write_history,
     fits_write_key,
     fits_write_pix,
+    fits_write_pixnull,
+    fits_write_subset,
+    fits_write_null_img,
     fits_write_record,
     fits_write_tdim,
     libcfitsio_version,
     cfitsio_typecode,
     bitpix_from_type,
     type_from_bitpix
 
+const READONLY = 0
+const READWRITE = 1
```

Comment:
Maybe these can be `Enum`s?  That'd allow us to make the signature more specific, too

Reply:
I agree an enum seems more appropriate

Reply:
I've changed this to use an enum in 031d41b. What do you recommend the signature should be? Currently it accepts ints as well as `READONLY/READWRITE` types.

Reply:
I think it might look better to do something like
```julia
@enum FileMode R=0 RW=1
function fits_open_file(filename, mode=R)
```
If we want to restrict users to only using the enum, not any integer, we just have to type `mode` to be `mode::FileMode=R`. I like this idea but I realize it might be a nuisance, thoughts?

I like the briefer names because they won't be exported and its nicer to call `CFITSIO.R` and `CFITSIO.RW`.

Reply:
I think restricting users to `FileMode` might be breaking, as integers should be valid arguments as well. Perhaps `Union{FileMode, Integer}` might be more appropriate. However I wonder if the specialization is worth it, since we explicitly convert the argument to `Int` anyway.

Regarding the shorter names, I wonder if readability is increased by the more verbose ones? Although perhaps `R` and `RW` would be familiar from their use in `FITSIO`, so that seems reasonable too.